piptools/scripts/compile.py

```diff
@@ -354,6 +363,7 @@ def cli(
             repository,
             prereleases=repository.finder.allow_all_prereleases or pre,
             clear_caches=rebuild,
+            cache_dir=cache_dir,
```

Comment:
```suggestion
            cache=DependencyCache(cache_dir),
```
I'd prefer this way, the benefit is that we can use different cache backends (mocked in tests for example)